webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueCompilationCache.scala

```diff
@@ -119,32 +140,66 @@ class TechniqueCompilationStatusService(
 
 /**
   * Technique compilation output needs to be saved in a cache (frequent reads, we don't want to get files on the FS every time).
-  * 
-  * This cache is a simple in-memory one which only saves errors, 
+  *
+  * This cache is a simple in-memory one which only saves errors,
   * so it saves only compilation output stderr message in case the compilation failed.
   * It notifies the lift actor on update of the compilation status.
-  * 
+  *
   * It is used mainly to get errors in the Rudder UI, and is updated when API requests to update techniques are made,
   * when technique library is reloaded
   */
 class TechniqueCompilationErrorsCache(
-    actor: SimpleActor[UpdateCompilationStatus],
-    ref:   Ref[Map[(BundleName, Version), EditorTechniqueError]]
+    actor:     SimpleActor[UpdateCompilationStatus],
+    reader:    ReadTechniqueCompilationStatusService,
+    errorBase: Ref[Map[(BundleName, Version), EditorTechniqueError]]
 ) extends WriteTechniqueCompilationStatusService {
```

Comment:
this one is more like a repo. It is the one that keep things synchronised. 
In a more complexe case, the service for gathering would be external and something else would call the repo. Here, we bundle the two together because it would be just overengineering to make a clean repos, and then a service to flatMap read and save. 

Same, the callback to actor could completely be implemented throught a call back on save - it's exactaly what it is, and clearly apparent in the for comprehension. But again, it would be totally over-engineered here. We just have to keep a note that the actor part could be decoupled from that repos for more unit responsibility if needed.